envoy/router/cluster_specifier_plugin.h

```diff
@@ -19,6 +19,17 @@ class ClusterSpecifierPlugin {
 public:
   virtual ~ClusterSpecifierPlugin() = default;
 
+  /**
+   * Validate if the clusters are valid in the cluster manager. This
+   * is optional.
```

Comment:
Minor: When we say _"This is optional."_, what exactly is optional? The result or calling the validation method itself?